File: app/src/main/java/com/planmanager/state/PlanState.java
package com.planmanager.state;

import com.planmanager.model.Plan;
import java.util.List;

public class PlanState {
    private final boolean isLoading;
    private final List<Plan> plans;
    private final String error;

    public PlanState(boolean isLoading, List<Plan> plans, String error) {
        this.isLoading = isLoading;
        this.plans = plans;
        this.error = error;
    }

    public static PlanState initial() {
        return new PlanState(false, List.of(), null);
    }

    public boolean isLoading() {
        return isLoading;
    }

    public List<Plan> getPlans() {
        return plans;
    }

    public String getError() {
        return error;
    }

    public PlanState copy(boolean isLoading, List<Plan> plans, String error) {
        return new PlanState(
                isLoading != null ? isLoading : this.isLoading,
                plans != null ? plans : this.plans,
                error != null ? error : this.error
        );
    }
}

File: app/src/main/java/com/planmanager/state/PlanState.kt
package com.planmanager.state

import com.planmanager.model.Plan

data class PlanState(
    val isLoading: Boolean,
    val plans: List<Plan>,
    val error: String?
) {
    companion object {
        fun initial() = PlanState(
            isLoading = false,
            plans = emptyList(),
            error = null
        )
    }

    fun copy(
        isLoading: Boolean? = null,
        plans: List<Plan>? = null,
        error: String? = null
    ): PlanState {
        return PlanState(
            isLoading = isLoading ?: this.isLoading,
            plans = plans ?: this.plans,
            error = error ?: this.error
        )
    }
}
